File: c/prog1KPG.c
/**
 * C portion of Project1
 * CSCI 265
 * Kaleb Gilson - kaleb.gilson@und.edu - 1/26/23
 */

#include <stdio.h>
#include <string.h>

void partOne();

void partTwo();

void partThree();

int main() {
    partOne();
    partTwo();
    partThree();
    return 0;
}

void partOne() {
    fflush(stdin); //Flushes input/output buffer
    int totalSeconds;

    printf("Seconds:");
    scanf("%d", &totalSeconds);

    int totalMinutes = totalSeconds/60;
    int hours = totalSeconds/3600; // define hours variable (This gets rounded up)
    int minutes = totalMinutes-(hours*60); // subtract hours (in minutes) from total minutes
    int seconds = totalSeconds-(minutes*60)-(hours*3600); // subtract minutes and hours (in seconds) from total seconds

    printf("This is %d hours %d minutes and %d seconds.", hours, minutes, seconds);
}

void partTwo() {
    fflush(stdin);
    int quarters, dimes, nickles, pennies;

    printf("\n\nQuarters:");
    scanf("%d", &quarters);

    printf("Dimes:");
    scanf("%d", &dimes);

    printf("Nickles:");
    scanf("%d", &nickles);

    printf("Pennies:");
    scanf("%d", &pennies);

    double total = 0; // Define the "total" variable

    // Multiply the amount of coins by the amount the coins are worth, add it to the total
    total+=(quarters*0.25);
    total+=(dimes*0.1);
    total+=(nickles*0.05);
    total+=(pennies*0.01);

    printf("This is equal to $%.2f", total); // Print the output
}

void partThree() {
    fflush(stdin);
    char firstName[50], lastName[50], address[100], city[50], state[11], zipCode[11];

    // Print the prompt for input
    printf("\n\nFirst name:");
    fgets(firstName, 49, stdin); // Take an int from the scanner to the quarters variable
    firstName[strcspn (firstName, "[\r\n]")] = 0; //Remove \r and \n from the string
    // Repeat for all input

    printf("Last name:");
    fgets(lastName, 49, stdin);
    lastName[strcspn (lastName, "[\r\n]")] = 0;

    printf("Address:");
    fgets(address, 100, stdin);
    address[strcspn (address, "[\r\n]")] = 0;

    printf("City:");
    fgets(city, 49, stdin);
    city[strcspn (city, "[\r\n]")] = 0;

    printf("State:");
    fgets(state, 10, stdin);
    state[strcspn (state, "[\r\n]")] = 0;

    printf("Zip code:");
    fgets(zipCode, 10, stdin);
    zipCode[strcspn (zipCode, "[\r\n]")] = 0;

    // Single print statement mailing address output
    printf("\n%s %s\n%s\n%s, %s  %s", firstName, lastName, address, city, state, zipCode);

    // Six print statement mailing address output
    printf("\n%s", firstName);
    printf(" %s", lastName);
    printf("\n%s", address);
    printf("\n%s", city);
    printf(", %s", state);
    printf("  %s", zipCode);
}

File: python/prog1KPG.py
def part1():
    total_seconds = int(input("Seconds: "))  # define seconds variable and pull an int from the command line.
    total_minutes = total_seconds // 60
    hours = total_seconds // 3600  # define hours variable (This gets rounded up)
    minutes = total_minutes - (hours * 60)  # subtract hours (in minutes) from total minutes
    seconds = total_seconds - (minutes * 60) - (hours * 3600)  # subtract minutes and hours (in seconds) from total seconds

    # Create a string to store the output
    print("This is " + str(hours) + " hours " + str(minutes) + " minutes and " + str(seconds) + " seconds.")


def part2():
    quarters = int(input("\nQuarters: "))  # Take an int from the scanner to the quarters variable
    # Repeat for all input

    dimes = int(input("Dimes: "))
    nickles = int(input("Nickles: "))
    pennies = int(input("Pennies: "))

    total = 0  # Define the "total" variable

    # Multiply the amount of coins by the amount the coins are worth, add it to the total
    total += (quarters * 0.25)
    total += (dimes * 0.1)
    total += (nickles * 0.05)
    total += (pennies * 0.01)

    # Round to the second decimal place
    round(total, 2)
    print("This is equal to $" + str(round(total, 2)))


def part3():
    first_name = input("\nFirst name: ")
    last_name = input("Last name: ")
    address = input("Address: ")
    city = input("City: ")
    state = input("State: ")
    zip_code = input("Zip Code: ")

    print(first_name + " " + last_name + "\n" + address + "\n" + city + ", " + state + "  " + zip_code)

    print(first_name + " ", end='')
    print(last_name, end='')
    print("\n" + address, end='')
    print("\n" + city + ", ", end='')
    print(state + "  ", end='')
    print(zip_code, end='')


part1()
part2()
part3()
